File: plugins/kotlin/j2k/new/tests/testData/newJ2k/types/nestedArrays.java
class A {
    public static int [][] sampleFunction(int a[][], int b[][]) {
        int n = a[0].length;
        int m = a.length;
        int p = b[0].length;
        int result[][] = new int[m][p];
        for (int i=0; i<m; i++){
            for(int j=0; j<p; j++){
                for (int k=0; k<n ; k++){
                    result[i][j] += a[i][k] * b[k][j];
                }
            }
        }
        return result;
    }
}

File: plugins/kotlin/j2k/new/tests/testData/newJ2k/types/nestedArrays.kt
internal object A {
    fun sampleFunction(a: Array<IntArray>, b: Array<IntArray>): Array<IntArray> {
        val n = a[0].size
        val m = a.size
        val p = b[0].size
        val result = Array(m) { IntArray(p) }
        for (i in 0 until m) {
            for (j in 0 until p) {
                for (k in 0 until n) {
                    result[i][j] += a[i][k] * b[k][j]
                }
            }
        }
        return result
    }
}
